main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/appconfigdata"
+)
+
+var verbose bool
+
+type AppConfigParams struct {
+	applicationID   string
+	environmentID   string
+	configProfileID string
+}
+
+func main() {
+	params := readFlags()
+
+	vars, err := getConfig(params)
+	if err != nil {
+		fmt.Printf("failed to get config: %s", err)
+		os.Exit(1)
+	}
+
+	args := flag.Args()
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Env = append(os.Environ(), vars...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if verbose {
+		fmt.Printf("running %q with args: %s and env: %s", args[0], args[1:], cmd.Env)
+	}
+	if err = cmd.Run(); err != nil {
+		os.Exit(2)
+	}
+}
+
+func readFlags() (params AppConfigParams) {
+	flag.StringVar(&params.applicationID, "app", "", "application identifier")
+	flag.StringVar(&params.environmentID, "env", "", "environment identifier")
+	flag.StringVar(&params.configProfileID, "config", "", "config profile identifier")
+	flag.BoolVar(&verbose, "v", false, "verbose output")
+	flag.Parse()
+
+	if params.applicationID == "" {
+		fmt.Printf("specify application identifier with --app flag")
+		os.Exit(1)
+	}
+	if params.environmentID == "" {
+		fmt.Printf("specify environment identifier with --env flag")
+		os.Exit(1)
+	}
+	if params.configProfileID == "" {
+		fmt.Printf("specify config profile identifier with --config flag")
+		os.Exit(1)
+	}
+
+	if flag.Arg(0) == "" {
```

Comment:
I think I like this more, but could keep it if you want.

```suggestion
	if flag.NArg() == 0 {
```